openingd/dualopend.c

```diff
@@ -1118,6 +1134,65 @@ static u8 *opening_negotiate_msg(const tal_t *ctx, struct state *state)
 			continue;
 		}
 
+		/* In theory, we're in the middle of an open/RBF, but
+		 * it's possible we can get some different messages in
+		 * the meantime! */
+		t = fromwire_peektype(msg);
+		switch (t) {
+		case WIRE_TX_SIGNATURES:
+			/* We can get these when we restart and immediately
+			 * startup an RBF */
+			handle_tx_sigs(state, msg);
+			continue;
+		case WIRE_FUNDING_LOCKED:
+			handle_funding_locked(state, msg);
+			return NULL;
+		case WIRE_SHUTDOWN:
+			handle_peer_shutdown(state, msg);
+			/* If we're done, exit */
+			if (shutdown_complete(state))
+				shutdown(state);
```

Comment:
Hmm, do we want to return msg to the caller here (if not shutdown_complete) ?  I expected a continue or a return NULL...

Reply:
Oops. Yes, we should return NULL.